packages/dashboard/src/components/FileUpload/index.tsx

```diff
@@ -0,0 +1,73 @@
+import { MaskColorVar } from '@dimensiondev/maskbook-theme'
+import { makeStyles, Typography } from '@material-ui/core'
+import { useEffect, useState } from 'react'
+import { File as FileIcon } from '../../../../icons/general/File'
+
+const useStyles = makeStyles(() => ({
+    root: {
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        position: 'relative',
+        height: '100%',
+        background: MaskColorVar.secondaryBackground,
+        borderRadius: 8,
+    },
+    container: {
+        textAlign: 'center',
+    },
+    file: {
+        position: 'absolute',
+        width: '100%',
+        height: '100%',
+        top: 0,
+        left: 0,
+        opacity: 0,
+        cursor: 'pointer',
+    },
+    text: {
+        color: MaskColorVar.textSecondary,
+        fontSize: 13,
+    },
+}))
+
+interface FileUploadProps {
+    width?: number
+    height?: number
+    readAsText?: boolean
+    onChange: (file: File, content?: string) => void
+}
```

Comment:
```suggestion
export type FileUploadProps = {
    width?: number
    height?: number
    readAsText: true
    onChange: (file: string) => void
} | {
    width?: number
    height?: number
    readAsText?: false
    onChange: (file: File) => void
}
```

Reply:
I think the original version is fine. We may need both file and content, as we may need to show file info like filename.

Reply:
```ts
export type FileUploadProps = {
    width?: number
    height?: number
    readAsText: true
    onChange: (file: File, file: string) => void
} | {
    width?: number
    height?: number
    readAsText?: false
    onChange: (file: File) => void
}
```